docs/src/architecture/09_architecture_decisions/0011-align-be-fe-dev-releases.md

```diff
@@ -0,0 +1,136 @@
+---
+    title: 0011  Align BE / FE unstable releases
+    adr:
+        author: Oleksandr Prokhorenko
+        created: 21-Oct-2025
+        status:  draft
+---
+
+## Context
+
+Our current deployment practices create instability and coordination problems:
+
+1. **Deployment process mismatch**: Frontend auto-deploys to `dev` on every `main` merge,
+but backend/gateway deploys are fully manual and controlled.
+This creates environment drift and makes it difficult to test compatible versions together.
+
+2. **Shared environment blocking**: The `dev` environment frequently gets blocked by specific versions (e.g., fund15)
+that teams want to stabilize and test, preventing other work from being deployed and tested.
+
+3. **Version traceability**: When issues occur, it's difficult to identify which version caused the problem
+because deployments aren't tied to explicit release tags.
+
+4. **Test environment accessibility**: Alternative test environments exist but require manual setup and
+training, making them not straightforward for general use.
+
+5. **Breaking changes in main**: Large breaking changes (like API V2) merged to `main` cause problems
+for both teams - backend loses ability to release from `main`, and frontend can't deploy without breaking changes.
+
+We agreed (Oct 13, 2025 discussion) to standardize on **releasing from tags only (not every merge)** for both BE and FE,
+with **feature-branch environments** for testing larger breaking changes before they land in `main`.
+
+## Assumptions
+
+- **OpenAPI** is (and remains) our single source of truth for public HTTP APIs.
+- Teams accept **spec-first** changes: spec updates land in PRs **before** code changes.
+- CI/CD can:
+  - compute **breaking diffs** between the current spec and the last released spec,
+  - run **spec conformance tests** against a running provider.
+- We can operate **channelized promotions** (unstable → testing → stable) and **preview envs** per PR/feature branch.
+- Weekly **release train** (lightweight) is acceptable; hotfixes remain possible.
+
+## Decision
+
+Adopt an **OpenAPI-driven, spec-first release model** with **tag-based deployments** to align BE(s) and FE:
+
+### Core Principle: Deploy from Tags, Not from Main
+
+Both backend and frontend will **only deploy tagged releases** to shared environments (`dev`, `qa`, `prod`).
+This ensures:
+- Explicit version tracking for troubleshooting
+- Controlled, coordinated releases between teams
+- Ability to keep environments stable during critical testing periods
+
+### 1. **Spec-first PRs**
+   - Any API change updates `openapi.yaml` in the backend repo first.
+   - FE + BE reviewers required on spec-changing PRs.
+
+### 2. **Breaking-change gate (OpenAPI diff)**
+   - CI compares PR's `openapi.yaml` to the **last released tag's** spec; **breaking diffs fail** the PR.
```

Comment:
I would want to clarify this part.
Am I correct that we want to bind the `OpenAPI` version to the released `cat-gateway` version ?
At this moment `OpenAPI` version reflects the version of the API definition, but does not necessarily reflects the version of the `cat-gateway` internal functionality.
So for example it could be that released [catalyst-openapi/v0.3.0](https://github.com/input-output-hk/catalyst-voices/releases/tag/catalyst-openapi%2Fv0.3.0) will not change for a long time, because the API definition would not change.
But in the meantime the actual internal functionality of the `cat-gateway` would be changed a lot (some bug fixes, stability improvements), and probably would not return the same results on the same API calls from the past.